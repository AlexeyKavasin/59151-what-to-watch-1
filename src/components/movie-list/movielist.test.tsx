import * as React from 'react';
import * as renderer from 'react-test-renderer';
import MovieList from './movielist';

const films = [
  {
    name: `Matrix`,
    poster: `https://es31-server.appspot.com/wtw/static/film/preview/matrix.jpg`,
    id: 0,
    genre: `Action`,
    preview_image: `https://es31-server.appspot.com/wtw/static/film/preview/matrix.jpg`,
    preview_video_link: `https://upload.wikimedia.org/wikipedia/commons/transcoded/b/b3/Big_Buck_Bunny_Trailer_400p.ogv/Big_Buck_Bunny_Trailer_400p.ogv.360p.webm`
  }
];

it(`Movie list renders correctly`, () => {
  const tree = renderer
    .create(<MovieList
      films={films}
      activeCard={-1}
      onMouseOver={jest.fn()}
      onMouseLeave={jest.fn()}
    />).toJSON();

  expect(tree).toMatchSnapshot();
});
