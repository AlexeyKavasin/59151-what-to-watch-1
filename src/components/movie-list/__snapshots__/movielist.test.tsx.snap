// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Movie list renders correctly 1`] = `
<div
  className="catalog__movies-list"
>
  <article
    className="small-movie-card catalog__movies-card"
    data-id={0}
    onMouseLeave={[Function]}
    onMouseOver={[Function]}
  >
    <div
      className="small-movie-card__image"
    >
      <video
        height="175"
        muted={true}
        poster="https://es31-server.appspot.com/wtw/static/film/preview/matrix.jpg"
        width="280"
      >
        <source
          src="https://upload.wikimedia.org/wikipedia/commons/transcoded/b/b3/Big_Buck_Bunny_Trailer_400p.ogv/Big_Buck_Bunny_Trailer_400p.ogv.360p.webm"
        />
      </video>
    </div>
    <h3
      className="small-movie-card__title"
    >
      <a
        className="small-movie-card__link"
        href="movie-page.html"
      >
        Matrix
      </a>
    </h3>
  </article>
</div>
`;
